Added a `where` method on the route class which can constraint a route its variables

DJWassink opened the pull request:
Still a WIP, maybe needs some discussion, but most certainly needs some readme ❤️

JBlaak commented on src/connect.ts:
```diff
@@ -27,6 +28,19 @@ export default function connect(app: Application,
+                // https://github.com/gotwarlost/istanbul/blob/master/ignoring-code-for-coverage.md#ignore-an-else-path
+                /* istanbul ignore else  */
+                if (constraints.hasOwnProperty(key)) {
+                    path = path.replace(key, key + constraints[key]);
```
Maybe a bit too naive: `route.get('/user/:identifier/comment/:id').where({id:'[0-9]+')`

DJWassink commented:
Most of the logic is now in https://github.com/JBlaak/Express-Torch/pull/24/files#diff-8265994ca17d8ef716f946876b8469ed

Also updated path-to-regex to version ^2.2.0, thus also removing the @types/path-to-regexp. Sadly updating path-to-regex to version 2.2.0 resulted in a yarn install conflict with @types/node (peer dep of path-to-regex maybe?), anyhow I also removed @types/node

JBlaak commented:
:heart_eyes: loving it, add description to the readme and we can merge it!

JBlaak commented on tests/torch.test.ts:
```diff
@@ -613,6 +614,180 @@ describe('Torch', function () {
+            Torch(app as Application, (router: Router) => {
+                router.delete('/:identifier/:id', method).where({
+                    'id': '(\\d+)',
+                    'identifier': '(\\d+)'
```
Erm, this doesn't test anything, we still do not known which one is inserted where
